### PR TITLE
Add support to import data from sav, dta, por, sas and stata using haven

### PR DESCRIPTION
**Pull request by javierluraschi**
This PR adds support to import data from sav, dta, por, sas and stata using haven. Refactored `FileOrUrlChooserTextBox` into `DataImportFileChooser` to reuse in sas options with support for flexbox.

screen shot 2016-01-28 at 8 30 40 am

### CODE REVIEW

File: src/gwt/src/org/rstudio/studio/client/common/dependencies/DependencyManager.java [hunk 1]
```diff
@@ -304,6 +304,39 @@ public void execute(Boolean succeeded)
       return deps.toArray(new Dependency[deps.size()]);
    }
    
+   public void withDataImportSAV(String userAction, final Command command)
+   {
+     withDependencies(
+        "Preparing Import from SAV",
```
Comment (jjallaire) [inline src/gwt/src/org/rstudio/studio/client/common/dependencies/DependencyManager.java]: I think we might want to refer to "SAV" as "SPSS". Does that make sense?

File: src/gwt/src/org/rstudio/studio/client/common/dependencies/DependencyManager.java [hunk 2]
```diff
@@ -304,6 +304,39 @@ public void execute(Boolean succeeded)
...
+                 command.execute();
+           }
+        }
+     );
+   }
+   
+   private ArrayList<Dependency> dataImportSavDependencies()
+   {
+      ArrayList<Dependency> deps = new ArrayList<Dependency>();
+      deps.add(Dependency.cranPackage("haven", "0.2.0"));
```
Comment (jjallaire) [inline src/gwt/src/org/rstudio/studio/client/common/dependencies/DependencyManager.java]: Note that the "haven" package requires R version 3.1 and the readr package requires R version 3.0. Technically RStudio supports as far back as R 2.11! That said, I don't think we need to make UI conditional on R >= 3 as a ton of our other UI including Shiny, R Markdown, Rcpp, devtools, etc. are all already predicated on R >= 3. I think we should however check for R >= 3.1 on the server and pass that up through `SessionInfo`, then check it in Application.initializeWorkbench and disable the haven import commands if we don't have R 3.1 or greater.

File: src/gwt/src/org/rstudio/studio/client/workbench/commands/Commands.cmd.xml [hunk 3]
```diff
@@ -2042,6 +2043,11 @@ well as menu structures (for main menu and popup menus).
         menuLabel="From _CSV..."
         windowMode="main"/>
 
+   <cmd id="importDatasetFromSAV"
+        label="Import Dataset from SAV..."
```
Comment (jjallaire) [inline src/gwt/src/org/rstudio/studio/client/workbench/commands/Commands.cmd.xml]: Import Dataset from SPSS....   ?

File: src/gwt/src/org/rstudio/studio/client/workbench/commands/Commands.cmd.xml [hunk 4]
```diff
@@ -364,6 +364,7 @@ well as menu structures (for main menu and popup menus).
             <cmd refid="importDatasetFromFile"/>
             <cmd refid="importDatasetFromURL"/>
             <cmd refid="importDatasetFromCSV"/>
+            <cmd refid="importDatasetFromSAV"/>
```
Comment (jjallaire) [inline src/gwt/src/org/rstudio/studio/client/workbench/commands/Commands.cmd.xml]: I think there should be a separator after import from text. Final menu might look something like this:

```
Import Dataset from Text...
-------------------------------------------
Import Dataset from Excel...
-------------------------------------------
Import Dataset from SPSS...
Import Dataset from SAS...
Import Dataset from Stata...
```

File: src/gwt/src/org/rstudio/studio/client/common/dependencies/DependencyManager.java (hunk 1 above)
Comment (javierluraschi) [inline src/gwt/src/org/rstudio/studio/client/common/dependencies/DependencyManager.java]: Yes, makes sense. SPSS is the product name, SAV the file format. Originally I thought of listing only extension names, but products is more clear in this case.

File: src/gwt/src/org/rstudio/studio/client/common/dependencies/DependencyManager.java (hunk 2 above)
Comment (javierluraschi) [inline src/gwt/src/org/rstudio/studio/client/common/dependencies/DependencyManager.java]: Good point, added checks and marking disabled when the R requirement is not met.

File: src/gwt/src/org/rstudio/studio/client/workbench/commands/Commands.cmd.xml (hunk 3 above)
Comment (javierluraschi) [inline src/gwt/src/org/rstudio/studio/client/workbench/commands/Commands.cmd.xml]: Changed.

**Comment by javierluraschi** [comment]
Added boilerplate code for the remaining dialogs. Here is how the complete menu looks like:

screen shot 2016-01-28 at 10 53 37 am

File: src/gwt/src/org/rstudio/studio/client/workbench/commands/Commands.cmd.xml (hunk 4 above)
Comment (javierluraschi) [inline src/gwt/src/org/rstudio/studio/client/workbench/commands/Commands.cmd.xml]: Makes sense.

**Comment by javierluraschi** [comment]
@jjallaire:
- Making them invisible now.
- I'm going with (2) if thats OK with you. (1) seems risky, (3) works as well but I would feel better keeping the try/catch block anyways.
- Makes sense, thanks!

**Pull request merged**